### PR TITLE
Features/mysql

### PR DESCRIPTION
**Pull request by vc-vishva**
- use mysql 
- connect mysql with node js project 
- perform mysql crud task

### CODE REVIEW

File: mysql task/src/app.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import express from "express";
+import route1 from "./routes/user.route.js";
```
Comment (gaurangviitorcloud) [inline mysql task/src/app.js:2]: Please change the naming convention what is route1,route2,route3?

File: mysql task/src/app.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import route3 from "./routes/permission.route.js";
+import route2 from "./routes/role.route.js";
+
+
+// import connection from "./db/connection.js";
+
+
+const app= express();
+app.use(express.json());
+app.use("/user", route1);
```
Comment (gaurangviitorcloud) [inline mysql task/src/app.js:12]: Consider using index.routes.js file so your app.js does not get big.

File: mysql task/src/app.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+app.use("/role", route2);
+app.use("/permission", route3);
+
+
+app.listen(3000, ()=> console.log("Server is running on ", 3000));
+
```
Comment (gaurangviitorcloud) [inline mysql task/src/app.js:18]: Why is their 19 empty lines  here ?

File: mysql task/src/routes/permission.route.js [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+      `);
+
+    res.send({ message: "success", data: permission });
+  } catch (error) {
+    res.status(500).send({ error });
+  }
+});
+//2
+
+route3.post("/add", async (req, res) => {
```
Comment (gaurangviitorcloud) [inline mysql task/src/routes/permission.route.js:24]: Your routes name should not include /add

File: mysql task/src/routes/permission.route.js [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+JOIN permission ON role.role= '${role}' AND permission.permission = '${permission}'`);
+
+    res.send({ message: "success", data: newdata });
+  } catch (error) {
+    res.status(500).send({ error });
+  }
+});
+//3
+
+ route3.delete("/delete", async (req, res) => {
```
Comment (gaurangviitorcloud) [inline mysql task/src/routes/permission.route.js:47]: /delete should not be part of route please improve route naming EVERYWHERE in project

File: mysql task/src/routes/role.route.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
+import express from "express";
+import query from "../db/query.js";
+
+const route2= express.Router();
```
Comment (gaurangviitorcloud) [inline mysql task/src/routes/role.route.js:4]: IMPORVE variable naming convention EVERYWHERE

**Review by gaurangviitorcloud — CHANGES_REQUESTED** [review]
- Please improve variable naming convention
- Please do not include unnecessary spaces after file

**Review by gaurangviitorcloud — CHANGES_REQUESTED** [review]
- Please improve variable naming convention
- Please do not include unnecessary spaces after file

File: mysql task/src/app.js (hunk 1 above)
Comment (vc-vishva) [reply mysql task/src/app.js:2]: ok,   I have change it.

File: mysql task/src/app.js (hunk 2 above)
Comment (vc-vishva) [reply mysql task/src/app.js:12]: ok

**Pull request closed without merging**